enzyme/Enzyme/GradientUtils.h

```diff
@@ -1263,24 +1273,27 @@ if (AtomicAdd) {
                          cast<PointerType>(ptr->getType())->getAddressSpace()));
     dif = BuilderM.CreateBitCast(dif, IntToFloatTy(dif->getType()));
   }
-  if (llvm::Triple(newFunc->getParent()->getTargetTriple()).getArch() ==
-          Triple::nvptx ||
-      llvm::Triple(newFunc->getParent()->getTargetTriple()).getArch() ==
-          Triple::nvptx64) {
-    if (dif->getType()->isFloatTy()) {
-      // auto atomicAdd =
-      cast<CallInst>(BuilderM.CreateCall(
-          Intrinsic::getDeclaration(newFunc->getParent(),
-                                    Intrinsic::nvvm_atomic_add_gen_f_sys,
-                                    {dif->getType(), ptr->getType()}),
-          {ptr, dif}));
-    } else if (dif->getType()->isDoubleTy()) {
+  if (false &&
+      (llvm::Triple(newFunc->getParent()->getTargetTriple()).getArch() ==
+           Triple::nvptx ||
+       llvm::Triple(newFunc->getParent()->getTargetTriple()).getArch() ==
+           Triple::nvptx64)) {
+
+    if (dif->getType()->isFloatTy() || dif->getType()->isDoubleTy()) {
       // auto atomicAdd =
-      cast<CallInst>(BuilderM.CreateCall(
-          Intrinsic::getDeclaration(newFunc->getParent(),
-                                    Intrinsic::nvvm_atomic_add_gen_f_sys,
-                                    {dif->getType(), ptr->getType()}),
-          {ptr, dif}));
+      if (ptr->getType()->getPointerAddressSpace() == 3) {
+        cast<CallInst>(BuilderM.CreateCall(
+            Intrinsic::getDeclaration(newFunc->getParent(),
+                                      Intrinsic::nvvm_atomic_add_gen_f_cta,
```

Comment:
Should this not be just the `llvm::atomic`?

Reply:
Code deleted.